merlin/models/torch/batch.py

```diff
@@ -277,6 +295,106 @@ def target(self, name: str = "default") -> torch.Tensor:
 
         raise ValueError("Batch has multiple target, please specify a target name")
 
+    def flatten_as_dict(self, inputs: Optional["Batch"]) -> Dict[str, torch.Tensor]:
+        """
+        Flatten features, targets, and sequences into a dictionary of tensors.
+
+        Each key should be prefixed with "features.", "targets.", "masks." or "lengths."
+
+        If inputs is provided, it includes all keys that are present in both self and inputs,
+        with the value from self being used when a key is present in both.
+
+        Parameters
+        ----------
+        inputs : Batch, optional
+            Another Batch object to include in the flattening process. The keys from the input
+            batch are also added with a prefix of "inputs.", by default None
+
+        Returns
+        -------
+        Dict[str, torch.Tensor]
+            A dictionary containing all the flattened features, targets, and sequences.
+        """
+        flat_dict: Dict[str, torch.Tensor] = self._flatten()
+        dummy_tensor = torch.tensor(0)
```

Comment:
We never use it, but we need to have it in order to keep the type of the `Dict[str, torch.Tensor]`. We could store the original value but that might take more memory, so that's why I added the dummy